src/WebJobs.Script/Config/ScriptSettingsManager.cs

```diff
@@ -2,27 +2,23 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Concurrent;
 using Microsoft.Extensions.Configuration;
 
 namespace Microsoft.Azure.WebJobs.Script.Config
 {
     public class ScriptSettingsManager
     {
         private static ScriptSettingsManager _instance = new ScriptSettingsManager();
-        private readonly ConcurrentDictionary<string, string> _settingsCache = new ConcurrentDictionary<string, string>();
-        private Func<IConfiguration> _configurationFactory = BuildConfiguration;
-        private Lazy<IConfiguration> _configuration = new Lazy<IConfiguration>(BuildConfiguration);
 
-        // for testing
-        public ScriptSettingsManager()
+        public ScriptSettingsManager(IConfiguration config = null)
```

Comment:
Since this instance is based on IConfiguration, it makes sense to me to flow this in as part of DI construction. That's why modified this ctor, and changed the instance registration in WebJobsServiceCollectionExtensions for this type.

Most usages that get the manager from the DI container will get the IConfiguration constructed instance. However, since this class also has a static singleton Instance, things are a bit messy. There are a lot of usages of the static everywhere, so I'm hesitant to do all the refactoring necessary to remove the static usages. Thoughts?